### DISCUSSION TITLE
Can’t download more than 11 minutes.

### DISCUSSION THREAD

**Discussion by bixupetit** [aclap-dev/video-downloadhelper#1314] [Help and Q&A]
### Your question

Hi team, 

I’ve recently tried downloading a few videos however the downloaded videos are all  just approximately 11 minutes long.
I’m using version v2.0.19 on Firefox 9.1.0.44 on macOS 15 (Sequoia).

Here’s an example of a video I’m trying to download: https://www.telecinco.es/entrevias/a-la-carta/20240902/entrevias-temporada-capitulo-online-completo_18_013351429.html

You might need a VPN connected to Spain to be able to reproduce it.

Thank you!

### Addon & CoApp version and details

version: 9.1.0.44
target: mozilla
channel: stable
lang: en-GB
coapp: {"found":true,"path":"/Applications/net.downloadhelper.coapp.app/Contents/MacOS/vdhcoapp","version":"2.0.19","new_version":false}
license: {"accepted":true,"email":"<removed>","key":"a767cabe5c8d4485"}
platform: aarch64 mac
UA: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:130.0) Gecko/20100101 Firefox/130.0
{
    "valid_license_message_has_been_displayed": true,
    "used_history_button": true,
    "download_directory": "/Users/lauracobo/dwhelper",
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive",
            "__serializer_value": "/Users/lauracobo/dwhelper"
        }
    },
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "open_count_store": 7,
    "successfull_dl": 21,
    "download_history": {
        "__serializer_tag": "map",
        "__serializer_value": []
    }
}

**Comment by Mattmjs** [comment]
Can you verify if you see this in the image for your example.
![DRM icon](https://github.com/user-attachments/assets/e1a9cd45-4e3c-4702-ad81-21e76fe8f170)

**Comment by bixupetit** [comment]
Hi Mattmjs,
This is what I see: https://imgur.com/UI0m4FH.png
I don't see the icon on your screenshot.

**Reply by Mattmjs** [reply]
Good, just making sure it's not protected by DRM. I want you to follow these next steps precisely.

1. Click the 3 dot button next to download.
2. Look down to details and click on it.
3. A page opens with more information about the media.
4. **Command + A**  to select all.
5. **Command + C** to copy information to the clipboard.
6. **Command + V** to paste the information here in your response.

**Comment by bixupetit** [comment]
Thank you, Mattmjs,

Here you go:

<html><head></head><body>
    
Tab | 1 (incognito: false)
-- | --
'Entrevías': Temporada 4, capítulo 1, en 'Telecinco'
https://www.telecinco.es/entrevias/a-la-carta/20240902/entrevias-temporada-capitulo-online-completo_18_013351429.html  with 8 headers
https://album.mediaset.es/eimg/2024/09/02/entrevias-25_8b09.jpg
true
https://rawvod.mediaset.es/bitmovin/387b8d9f-f483-4342-a2bf-7d7d7ef60bba/387b8d9f-f483-4342-a2bf-7d7d7ef60bba.2823654.mp4
https://rawvod.mediaset.es/bitmovin/387b8d9f-f483-4342-a2bf-7d7d7ef60bba/387b8d9f-f483-4342-a2bf-7d7d7ef60bba.2823654.mp4
none
Mp4 b:HTTPMedia p:non-adaptative unknowns
unknown at -1, fps: ?, q:?, no dim


    
  

</body></html>

**Reply by bixupetit** [reply]
I see the table does not display properly here, so I'm adding a screenshot as well: https://i.imgur.com/Xau8ipL.png

**Reply by Mattmjs** [reply]
Thanks, you followed my steps correctly. It's not chunked media but a standard mp4 media stream.
I can normally view these outside of the site in a new tab but the telecinco site has made it so that this doesn't work,
therefore I can't test the download myself unless I get a vpn and go to your link.

Since it is an mp4 media, which method did you choose to download. Via browser download or Companion app ?
Either will work or sometimes the companion app is required.

**Reply by bixupetit** [reply]
Thanks, yes, I'm actually in the UK and need a VPN for those.
I was using the browser extension.

**Reply by Mattmjs** [reply]
What I mean is in More settings, there is a setting that decides if the browser is used for the download. I have it checked.
If it isn't checked then the companion app will download it. You should experiment with that and try either way and see if it helps before I consider testing it with a vpn myself. 
See my screenshot:
![VDH settings](https://github.com/user-attachments/assets/e0813a5f-413b-40b9-80dd-5bc24b2be71b)

**Comment by bixupetit** [comment]
Hi Mattmjs,

Thank you for clarifying, I have tried with and without the options and I'm getting the same, an output video of 11-ish minutes.

**Comment by Mattmjs** [comment]
Ok I spent enough time on this and getting nowhere , about 1 hour 30 minutes after switching to a Spanish location.
As you say I was only getting over 11 minutes before it would stop, by 16% the speed of the download went down to nothing.
I thought I would get an mp4 file like what your media information showed but was HLS, maybe the mp4 file was for an advertisement.
I have Ublock Origin, you don't have any adblocker.

![spanish video](https://github.com/user-attachments/assets/3a7495a8-121b-4be4-a500-7c18d02da967)

I tried ffmpeg by itself and there was loads of stuff going on, perhaps this is what is causing it to stop working.
The files itself don't seem to be recorded from the start which doesn't make sense.

One last thing I tried was switching the master manifest extension to .mpd which looks like it could work but it was quite slow,
probably throttled so stopped the download. This did create file from the start.

I've attached a network recording for further investigation.

[www.telecinco.es_Archive [24-10-01 17-43-12].zip](https://github.com/user-attachments/files/17201635/www.telecinco.es_Archive.24-10-01.17-43-12.zip)

**Reply by bixupetit** [reply]
Hi Mattmjs,

Thank you for investigating the issue. I'm not entirely sure about the file you are sharing with me as I see a .har format that I don't recognise. Should I do something with it?

Just for reference, the same exact thing happens with this other video: https://www.cuatro.com/quien-quiere-casarse-con-mi-hijo/a-la-carta/20240910/quien-quiere-casarse-con-mi-hijo-programa-en-video_18_013406570.html

Best,

**Reply by Mattmjs** [reply]
Just delete it, it's not for you but for the dev @paulrouget  and any tech heads to investigate further and find the issue.

**Reply by Mattmjs** [reply]
Same issue on that site like the other. Network activity below.
[www.cuatro.com_Archive [24-10-01 19-48-09].zip](https://github.com/user-attachments/files/17203016/www.cuatro.com_Archive.24-10-01.19-48-09.zip)

**Comment by bixupetit** [comment]
Thank you for clarifying, Mattmjs!

**Reply by bixupetit** [reply]
Hi team,

Have you had the chance to look into this issue?

Thank you,
Laura

**Reply by Mattmjs** [reply]
I don't know if it has been looked at or fixed yet but I have a workaround. The only problem is it's severely throttled.
Does this sound like it's worth trying or not ?

**Comment by bixupetit** [comment]
I confirm the issue is still the same so I'm happy to learn the workaround if any.

**Comment by Mattmjs** [comment]
This is just a suggestion since as a I say the speed is really slow but on your sites telecinco.es and cuatro.com try something
around 30 to 45 minutes range to see if it works. 

What you want to do is go to the video and open the developer tools by pressing **Command + Shift + E**. But don't play it yet.
This should open the Network. Filter the network for **m3u8** then click play on the video. A few links should appear.
You also want to clear the VDH menu and make it empty as those download options don't work.

I've underlined the link that needs to be changed in the screenshot, by right clicking on it and choosing Edit and Resend.

![spanish site](https://github.com/user-attachments/assets/6faf4560-e6e9-4c2a-ae69-1506a8e630f5)

After that click on the link as shown in the image, the part that will be edited is m3u8.

![edit and resend](https://github.com/user-attachments/assets/c7ee4121-759b-4c32-8cb8-ee17b75bc643)

Now change it to mpd as shown in the image then click send. The developer tools can be closed and you should have a new
download option in VDH.

![edit and resend 1](https://github.com/user-attachments/assets/3f08dcec-c4b3-4b73-bf15-ecb4f3478aac)

Instead of the HLS options it should show as MPD.
![spanish site download menu](https://github.com/user-attachments/assets/3bc95185-4486-4194-b77f-cd8b99776b84)

**Comment by bixupetit** [comment]
Thank you @Mattmjs for suggesting this workaround, I have now tested and it does work, however, the download is of very low quality, only 640x360. I tried twice to make sure I was selecting the 1080p option to no avail.
Can something be done to download the higher-quality versions?
Thanks again!

**Reply by Mattmjs** [reply]
I don't know why it does that, I tested in ffmpeg which is what the extension uses to download the file.
I specified one of the 1080p options and it worked getting the higher quality but for some reason not working in VDH.
That's probably your next option if you want something better than 360p. Have you used ffmpeg before ?

**Reply by bixupetit** [reply]
I have never used ffmpeg before. Is that another tool used to download videos or just another file extension?

**Reply by Mattmjs** [reply]
It's another tool to download videos, VDH uses it to perform the download but when used on it's own it might seem
intimidating as it's run from the command prompt on Windows or the terminal on Mac and Linux.
But once you know what commands to use it's not too bad.

**ffprobe "mpd link"** will bring up the formats for your video. It's the same link I showed that you change from m3u8 to mpd.
See below, You can see that stream 0:4 and 0:5 are 1080 and stream 0:6 is the audio. 0:5 is higher quality than 0:4

**ffmpeg -i "mpd link" -codec: copy -map 0:5 -map 0:6 "file name.mp4" will download 1080p with audio.

Input #0, dash, from 'https://vbdk.mediaset.es/orivod/vod/bitmovin/34/08/3408cdf7-9fff-43b2-8fec-a318e02706a7/main.ism/picky.mpd?hdnts=st=1732360495~exp=1732374895~acl=/*~hmac=63bccf1c1cdc6c27b083b8d97d85b1d408a50b0b744e56ec8222f959be7c0182':
  Duration: 01:47:20.00, start: 0.000000, bitrate: 0 kb/s
  Program 0
  Stream #0:0: Video: h264 (Main) (avc1 / 0x31637661), yuv420p, 640x360 [SAR 1:1 DAR 16:9], 188 kb/s, 600 fps, 25 tbr, 600 tbn (default)
      Metadata:
        variant_bitrate : 249656
        id              : video=249656
  Stream #0:1: Video: h264 (Main) (avc1 / 0x31637661), yuv420p, 768x432 [SAR 1:1 DAR 16:9], 318 kb/s, 600 fps, 25 tbr, 600 tbn (default)
      Metadata:
        variant_bitrate : 402592
        id              : video=402592
  Stream #0:2: Video: h264 (Main) (avc1 / 0x31637661), yuv420p, 1024x576 [SAR 1:1 DAR 16:9], 517 kb/s, 600 fps, 25 tbr, 600 tbn (default)
      Metadata:
        variant_bitrate : 632488
        id              : video=632488
  Stream #0:3: Video: h264 (Main) (avc1 / 0x31637661), yuv420p, 1600x900 [SAR 1:1 DAR 16:9], 976 kb/s, 600 fps, 25 tbr, 600 tbn (default)
      Metadata:
        variant_bitrate : 1174288
        id              : video=1174288
  Stream #0:4: Video: h264 (Main) (avc1 / 0x31637661), yuv420p, 1920x1080 [SAR 1:1 DAR 16:9], 1901 kb/s, 600 fps, 25 tbr, 600 tbn (default)
      Metadata:
        variant_bitrate : 2166232
        id              : video=2166232
  Stream #0:5: Video: h264 (Main) (avc1 / 0x31637661), yuv420p, 1920x1080 [SAR 1:1 DAR 16:9], 3455 kb/s, 600 fps, 25 tbr, 600 tbn (default)
      Metadata:
        variant_bitrate : 3992664
        id              : video=3992664
  Stream #0:6(en): Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, stereo, fltp, 127 kb/s (default)
      Metadata:
        variant_bitrate : 125368
        id              : audio_eng=125368

**Comment by julifos** [comment]
I've been reading this thread and it's related to the mine. Different website, same streaming server (mediaset):
https://github.com/aclap-dev/video-downloadhelper/discussions/1497

Both "Edit and Resend" and _ffprobe_ methods work for me. Any way to include this fix in VDH?
